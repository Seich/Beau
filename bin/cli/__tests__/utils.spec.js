const utils = require('../utils.js');

jest.mock('fs');

describe('utils', () => {
    describe('openConfigFile', () => {
        it('should read and parse the given configuration file.', () => {
            expect(utils.openConfigFile('beau.yml')).toMatchSnapshot();
        });

        it('should throw if given not given a file', () => {
            expect(() => utils.openConfigFile('not-a-file.yml')).toThrow();
        });
    });

    describe('loadConfig', () => {
        it('should load load the config onto Beau', () => {
            let beau = utils.loadConfig('beau.yml');
            expect(beau.config).toMatchSnapshot();
        });

        it('should load params onto the environment', () => {
            let beau = utils.loadConfig('beau.yml', [
                'HELLO=WORLD',
                'BYE=MARS'
            ]);
            expect(beau.config.ENVIRONMENT).toMatchSnapshot();
        });
    });
});
