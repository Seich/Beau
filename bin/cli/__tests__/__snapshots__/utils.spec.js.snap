// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils loadConfig should load load the config onto Beau 1`] = `
Config {
  "COOKIEJAR": false,
  "DEFAULTS": Object {},
  "ENDPOINT": "https://example.org/",
  "ENVIRONMENT": Object {
    "_": Object {},
  },
  "HOSTS": Array [],
  "PLUGINS": Plugins {
    "context": Object {
      "createReadStream": [Function],
    },
    "registry": Object {
      "dynamicValues": Array [
        Object {
          "fn": [Function],
          "name": "createReadStream",
        },
      ],
      "postRequestModifiers": Array [],
      "preRequestModifiers": Array [],
    },
  },
  "REQUESTS": Array [
    Object {
      "ALIAS": "anything",
      "COOKIEJAR": false,
      "ENDPOINT": "https://example.org/",
      "PAYLOAD": Object {
        "name": "$env.params.name",
      },
      "REQUEST": "GET /anything",
    },
  ],
  "VERSION": 1,
  "configKeys": Array [
    "VERSION",
    "ENDPOINT",
    "PLUGINS",
    "DEFAULTS",
    "ENVIRONMENT",
    "HOSTS",
    "COOKIEJAR",
  ],
}
`;

exports[`utils loadConfig should load params onto the environment 1`] = `
Object {
  "_": Object {
    "BYE": "MARS",
    "HELLO": "WORLD",
  },
}
`;

exports[`utils openConfigFile should read and parse the given configuration file. 1`] = `
Object {
  "GET /anything": Object {
    "alias": "anything",
    "payload": Object {
      "name": "$env.params.name",
    },
  },
  "endpoint": "https://example.org/",
  "version": 1,
}
`;
